hack/generate.sh

```diff
@@ -62,7 +82,42 @@ run_target() {
       $REPO_ROOT/hack/update-protobuf.sh
       ;;
     codegen)
-      $REPO_ROOT/hack/update-codegen.sh --"$MODE"
+      IFS=' ' read -ra available_options <<< "${AVAILABLE_CODEGEN_OPTIONS[@]}"
+      local which=$WHICH
+      if [[ -z "$which" ]]; then
+        which=("${available_options[@]}")
+        valid_options=("${available_options[@]}")
+      else
+        valid_options=()
+        invalid_options=()
+        
+        IFS=' ' read -ra WHICH_ARRAY <<< "$which"
+        for option in "${WHICH_ARRAY[@]}"; do
+            valid=false
+        
+            for valid_option in "${available_options[@]}"; do
+                if [[ "$option" == "$valid_option" ]]; then
+                    valid=true
+                    break
+                fi
+            done
+        
+            if $valid; then
+                valid_options+=("$option")
+            else
+                invalid_options+=("$option")
+            fi
+        done
+        
+        if [[ ${#invalid_options[@]} -gt 0 ]]; then
+            printf "Skipping invalid options: %s, Available options are: %s\n\n" "${invalid_options[*]}" "${available_options[*]}"
+        fi
```

Comment:
I guess we would rather vote to exit and fail if there are invalid options specified. I think of the the cli tools do this, right? It makes easier to debug things. Otherwise you have to notice this message in the output to find out that you have a typo or a wrong option. If we fail right away, then the caller will have to check and fix that passed options.